### PR TITLE
Alexander Msumba_ South Africa_ javascript_ core _1_ week 2

### PR DESCRIPTION
**Pull request by MSUMBA-art**
# Your Details

- Your Name: Alexander
- Your City: Cape Town- South Africa
- Your Slack Name: Alexander Msumba

# Homework Detailsk

- Module:  JavaScript-Core-1
- Week: 2

### CODE REVIEW

**Review by SerialCarKey — CHANGES_REQUESTED** [review]
Hi Alexander, Once again . super clean code .

i've just noticed some small things to review on the Mandatory exercises

There are slight Syntax errors within the Fix-Functions.JS & the Function-Creation .
 
such as the Identifier 'sortedLetters' has already been declared syntax error within Fix-Functions.

Other than that, Great Work :D

File: exercises/D-predicates/exercise.js [hunk 1]
```diff
@@ -6,10 +6,14 @@
 */
 
 // Finish the predicate function to test if the passed number is negative (less than zero)
-function isNegative(number) {}
+function isNegative(number) {
+  return typeof number === "value";
```
Comment (he-man-david) [inline exercises/D-predicates/exercise.js:10]: what is typeof number here and what is question asking?

File: exercises/E-conditionals/exercise.js [hunk 2]
```diff
@@ -8,6 +8,12 @@
 
 var name = "Daniel";
 var danielsRole = "mentor";
+if (name) {
```
Comment (he-man-david) [inline exercises/E-conditionals/exercise.js:11]: how can we compare to see if daniel is mentor?

File: exercises/G-conditionals-2/exercise-1.js [hunk 3]
```diff
@@ -6,7 +6,14 @@
   - if number is more or equal to zero, return the word "positive"
 */
 
-function negativeOrPositive(number) {}
+function negativeOrPositive(number) {
+  if (number < 0) {
+    return "negative"; 
+
+    } else if (number >= 0){
```
Comment (he-man-david) [inline exercises/G-conditionals-2/exercise-1.js:13]: do we need else if?

File: exercises/G-conditionals-2/exercise-2.js [hunk 4]
```diff
@@ -7,7 +7,16 @@
 
 */
 
-function studentPassed(grade) {}
+function studentPassed(grade) {
+  if (grade < 50) {
+    return "failed";
+  } else if (grade > 50) {
+      return "passed";
+  }else if(grade => 50) {
```
Comment (he-man-david) [inline exercises/G-conditionals-2/exercise-2.js:15]: do we need 3 conditions here?

File: exercises/G-conditionals-2/exercise-3.js [hunk 5]
```diff
@@ -2,13 +2,24 @@
...
-  - if the mark is 80 or higher then the grade is "A"
+  - if the mark is 80 or higher than the grade is "A"
   - if the mark is lower than 80 and greater than 60 then the grade is "B"
   - if the mark is 60 or lower but no lower than 50 then the grade is "C"
   - Otherwise the grade is "F"
 */
 
-function calculateGrade(mark) {}
+function calculateGrade(mark) {
+  if (mark >= 80) {
```
Comment (he-man-david) [inline exercises/G-conditionals-2/exercise-3.js:12]: great job!

File: mandatory/1-fix-functions.js [hunk 6]
```diff
@@ -13,17 +13,19 @@
 function mood() {
   let isHappy = true;
 
+function mood(isHappy) {
   if (isHappy) {
     return "I am happy";
   } else {
     return "I am not happy";
   }
+  return "I am not happy";
```
Comment (he-man-david) [inline mandatory/1-fix-functions.js:22]: will this line ever execute?

File: mandatory/2-function-creation.js [hunk 7]
```diff
@@ -5,7 +5,14 @@ Write a function that:
 - removes any forward slashes (/) in the strings
 - makes the string all lowercase
 */
-function tidyUpString(strArr) {}
+function tidyUpString(strArr) {
+  for (let i = 0; i < strArr.length; i++) {
+  strArr[i] = strArr[i].toLowerCase();
```
Comment (he-man-david) [inline mandatory/2-function-creation.js:10]: can we reduce calling strArr[i] so many times?

File: mandatory/2-function-creation.js [hunk 8]
```diff
@@ -15,14 +22,26 @@ Complete the function to check if the variable `num` satisfies the following req
...
 The function must NOT change the original array, arr.
 */
 
 function remove(arr, index) {
+  let newArr = [];
+  for (let i=0; i<arr.length; i++){
+    newArr.push(arr[i]);
+  };
+  newArr.splice(index, 1);
   return; // complete this statement
```
Comment (he-man-david) [inline mandatory/2-function-creation.js:45]: are we actually returning anything here?

File: mandatory/2-function-creation.js [hunk 9]
```diff
@@ -15,14 +22,26 @@ Complete the function to check if the variable `num` satisfies the following req
...
+}
 
 /* 
 Write a function that returns a copy of the given array arr, but with the element at the given index, index removed.
 The function must NOT change the original array, arr.
 */
 
 function remove(arr, index) {
+  let newArr = [];
+  for (let i=0; i<arr.length; i++){
```
Comment (he-man-david) [inline mandatory/2-function-creation.js:41]: we can also do this let newArr = [...arr]; Can we do this without creating a new array? Try using a filter! https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

**Comment by MSUMBA-art** [comment]
Thanks, will go through again

**Pull request closed without merging**